policyengine_us/variables/gov/states/ny/nysdps/ny_clean_heat_incentive.py

```diff
@@ -0,0 +1,103 @@
+        description = tax_unit("ny_clean_heat_description_category", period) #var6
+
+        # calc uncapped incentive
+        uncapped_incentive =  select(
+            [
+                # residential -> ashp
+                family_type == family_type.possible_values.RESIDENTIAL and heat_pump == heat_pump.possible_values.ASHP,
+                # residential -> gshp
+                family_type == family_type.possible_values.RESIDENTIAL and heat_pump == heat_pump.possible_values.GSHP,
+                # multifamily
+                family_type == family_type.possible_values.MULTIFAMILY,
+            ],
+            [
+                p.residential_ashp_amount[dac][home][heat_pump_type],
+                p.residential_gshp_amount[dac],
+                p.multifamily_amount[heat_pump][building][description],
+            ],
+        )
+
+        # multipy uncapped incentive by MMBtu/dwelling_unit (if necessary)
+        mmbtu = tax_unit('ny_clean_heat_mmbtu', period)
+
+        max_unit = p.dwelling_unit_cap
+        uncapped_dwelling_unit = tax_unit("ny_clean_heat_dwelling_unit", period)
+        dwelling_unit = min(max_unit, uncapped_dwelling_unit)
+
+        uncapped_incentive = select(
+            [
+                # residential
+                family_type == family_type.possible_values.RESIDENTIAL,
+                # multifamily -> C2 or C6A
+                description == description.possible_values.C2 or description == description.possible_values.C6A,
+                # multifamily -> C4, C4A, C6, C10
+                family_type == family_type.possible_values.MULTIFAMILY and (
+                    description == description.possible_values.C4 or \
+                    description == description.possible_values.C4A1 or \
+                    description == description.possible_values.C4A2 or \
+                    description == description.possible_values.C6 or \
+                    description == description.possible_values.C10),
+            ],
+            [
+                # N/A
+                uncapped_incentive,
+                # multipy by dwelling_unit
+                uncapped_incentive * dwelling_unit,
+                # multipy by mmbtu
+                uncapped_incentive * mmbtu,
+            ],
+        )
+
+        # calc cap amount
+        rate = select(
+            [
+                family_type == family_type.possible_values.RESIDENTIAL,
+                family_type == family_type.possible_values.MULTIFAMILY,
+            ],
+            [
+                p.residential_rate[dac],
+                p.multifamily_rate,
+
+            ],
+        )
+        project_cost = tax_unit("ny_clean_heat_project_cost", period)
+        cap = project_cost * rate
+
+        # calc capped incentive
+        family_type_bool = family_type == family_type.possible_values.RESIDENTIAL
+        incentive = where(family_type_bool, min(uncapped_incentive, cap), min(uncapped_incentive, cap, p.cap))
```

Comment:
separate each min statement into its own variables proceeding the where statement